### ISSUE TITLE
Meteor.LoggingIn return true when not loggedIn, and Meteor.user() is not saved to AsyncStorage

### ISSUE THREAD

**Issue by radiegtya** [inProgress-team/react-native-meteor#19]
Meteor.LoggingIn return true when not loggedIn, but when loggedIn it return false.
And the most blocker thing is Meteor.user() is not saved to asyncStorage. So when I am refreshing the app, the Meteor.user() is not there anymore

**Comment by Mokto** [comment]
Yes see [meteor docs for loggingIn](http://docs.meteor.com/#/full/meteor_loggingin) but it is what is expected ;) *LoggingIn* returns *true* if logging is in progress.

When you app is launched, React Native Meteor has to re-sign in. So it checks if a token exists (AsyncStorage), and sign in with it.

Am I being clear ?

**Comment by radiegtya** [comment]
Yea thanks for confirm. btw how about the Meteor.user() that undefined although already lloggedin when rn is refreshed

**Comment by Mokto** [comment]
Yeah Meteor.user() get defined only when re-signed in. Is it bad in your opinion ?

**Comment by radiegtya** [comment]
Yeah, meteor default behaviour are using it as the saved object to be used
when a user already loggedin. So can you explain how is it now to detect
that user already logged in and get their data everytime its needed? Btw
your Meteor.userId() also not saved to storage, and that should be saved yo
storage too

**Comment by radiegtya** [comment]
the diagram should be like this.

Login with password > success > save Meteor.user() obj and Meteor.userId string to asyncStorage by default. > app closed/refreshed on emulator > Meteor.user() and Metror.userId should still exist so i can redirect router to main page rather than login page for example

**Comment by radiegtya** [comment]
Sorry I am misunderstand your concept, it's already there if I call Meteor.user() or Meteor.userId() inside getMeteorData method, and the behaviour is already correct. But why its called so many time? and also with undefined data included. Is it possible to called only once when ready?

<img width="1192" alt="meteoruserid" src="https://cloud.githubusercontent.com/assets/2161622/13677457/a87baa82-e71c-11e5-9fd7-888f736f2510.png">

**Comment by Mokto** [comment]
Actually the Meteor.user is never saved in localstorage, only userId is ;)

![capture d ecran 2016-03-14 a 09 55 51](https://cloud.githubusercontent.com/assets/5103928/13746791/741048f0-e9cb-11e5-9564-c9c215f76675.png)

I will add this to asyncStorage soon, but it is not a priority in my opinion.

**Comment by radiegtya** [comment]
this is very useful when needing something like this, after logged In

Meteor.userId() use case on every part of method:
```javascript
startMeteorSubscriptions(){
  Meteor.subscribe('todos', {createdBy: Meteor.userId()}); //but currently Meteor.userId() can't be accessed like this
}
```

Meteor.user() use case
```javascript
render(){
   const user = Meteor.user();

   return (
     <NavigationBar
     title="My Profile"
     renderLeftButton={<Text>user.name</Text>}/>
   );
}
```

there are still so many case need this to be accessed Synchronously in any method provided by component. For now I can doing this by saving manually to AsyncStorage, and get data via props/state, and that's not so clear solution.

**Comment by Mokto** [comment]
I agree with you Meteor.userId() use case, but in you Meteor.user() use case, why don't you just :

```javascript
getMeteorData() {
   return { user: Meteor.user() };
}
render(){
   const { user } = this.data;

   return (
     <NavigationBar
     title="My Profile"
     renderLeftButton={<Text>user.name</Text>}/>
   );
}
```

??

Meteor is not saving Meteor.user(), so I'm do not really like adding Meteor.user() to AsyncStorage...

**Comment by radiegtya** [comment]
yeah the Meteor.user is not something that important, I just think that it was easier like that. But the Meteor.userId() in subscriptions etc, i think should be done :)

**Comment by Mokto** [comment]
Looking into this this week-end ;)

**Comment by Mokto** [comment]
Sorry but I couldn't add Meteor.userId() to AsyncStorage, because Async is asynchronous.

But now you can do that :

```javascript
startMeteorSubscriptions(){
  Meteor.subscribe('todos', {createdBy: Meteor.userId()});
}
```

Reactive variables are now accepted in startMeteorSubscriptions ;)

In Beta13

**Comment by radiegtya** [comment]
I'll give this a try. Hope this not make any memory leak, because reactive variable always called too many time and cannot be stopped

**Comment by radiegtya** [comment]
working like charm! I can get Meteor.userId() from everywhere and it also able to help hack something like onReady in subscriptions etc. 

Thanks @Mokto
